cli/azd/pkg/containerapps/container_app.go

```diff
@@ -118,105 +161,151 @@ func (cas *containerAppService) AddRevision(
+			isProvisioned = true
+		}
+
+		switch *revision.Properties.HealthState {
+		case armappcontainers.RevisionHealthStateUnhealthy:
+			return fmt.Errorf("revision '%s' is unhealthy", revisionName)
+		case armappcontainers.RevisionHealthStateNone, armappcontainers.RevisionHealthStateHealthy:
+			isHealthy = true
+		}
+
+		if isProvisioned && isHealthy {
+			result = &revision
+			return nil
+		}
+
+		return fmt.Errorf("revision '%s' is an unknown state", revisionName)
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("validating revision '%s': %w", revisionName, err)
+	}
+
+	return result, nil
 }
 
-func (cas *containerAppService) setTrafficWeights(
+func (cas *containerAppService) ShiftTrafficToRevision(
 	ctx context.Context,
 	subscriptionId string,
 	resourceGroupName string,
 	appName string,
-	containerApp *armappcontainers.ContainerApp,
 	revisionName string,
 ) error {
+	containerApp, err := cas.Get(ctx, subscriptionId, resourceGroupName, appName)
+	if err != nil {
+		return fmt.Errorf("getting container app: %w", err)
+	}
+
 	containerApp.Properties.Configuration.Ingress.Traffic = []*armappcontainers.TrafficWeight{
 		{
 			RevisionName: &revisionName,
 			Weight:       convert.RefOf[int32](100),
 		},
 	}
 
-	err := cas.updateContainerApp(ctx, subscriptionId, resourceGroupName, appName, containerApp)
+	err = cas.updateContainerApp(ctx, subscriptionId, resourceGroupName, appName, containerApp)
 	if err != nil {
 		return fmt.Errorf("updating traffic weights: %w", err)
 	}
 
 	return nil
 }
 
-func (cas *containerAppService) getContainerApp(
+func (cas *containerAppService) syncSecrets(
 	ctx context.Context,
 	subscriptionId string,
 	resourceGroupName string,
 	appName string,
+	containerApp *armappcontainers.ContainerApp,
 ) (*armappcontainers.ContainerApp, error) {
+	// If the container app doesn't have any secrets, we don't need to do anything
+	if len(containerApp.Properties.Configuration.Secrets) == 0 {
+		return containerApp, nil
+	}
+
 	appClient, err := cas.createContainerAppsClient(ctx, subscriptionId)
 	if err != nil {
 		return nil, err
 	}
 
-	containerAppResponse, err := appClient.Get(ctx, resourceGroupName, appName, nil)
+	// Copy the secret configuration from the current version
+	// Secret values are not returned by the API, so we need to get them separately
+	// to ensure the update call succeeds
+	secretsResponse, err := appClient.ListSecrets(ctx, resourceGroupName, appName, nil)
 	if err != nil {
-		return nil, fmt.Errorf("getting container app: %w", err)
+		return nil, fmt.Errorf("listing secrets: %w", err)
 	}
 
-	return &containerAppResponse.ContainerApp, nil
+	secrets := []*armappcontainers.Secret{}
+	for _, secret := range secretsResponse.SecretsCollection.Value {
+		secrets = append(secrets, &armappcontainers.Secret{
+			Name:  secret.Name,
+			Value: secret.Value,
+		})
+	}
+
+	containerApp.Properties.Configuration.Secrets = secrets
+
+	return containerApp, nil
```

Comment:
I'm a little confused here.

It looks like we are directly updating the `containerApp` which is the input parameter and then we use it as the output. Is that the intention?
Usually, when a method takes `foo` as input and returns `foo`, the expectation is that the input `foo` is not mutated